core/vm/vmcontext/caller.go

```diff
@@ -18,6 +18,15 @@ type evmCaller struct {
 	dontMeterGas bool
 }
 
+func NewTestingEVMCaller(vmConfig vm.Config, vmContext vm.Context, chainConfig *params.ChainConfig, state vm.StateDB) vm.EVMCaller {
+	return &evmCaller{
+		vmConfig:    vmConfig,
+		vmContext:   vmContext,
+		chainConfig: chainConfig,
+		state:       state,
+	}
+}
```

Comment:
won't work, since `evmCaller` is private, and the test is not in this pkg, so needs a public facing funciton

Reply:
I had a look at the test and I can see that the caller is used to lookup the gasPriceMinimum, but that fails because the registry is not deployed. As a result the fallback gas price minimum is returned. GetGasPriceMinimum also checks for a nil evmCaller and if so will return the fallback value as well.

So in this case I think it would be neater to just use a nil caller in the test and then there is no need for to add this test constructor.

Reply:
done!